helpers/TestCenterHelper.php

```diff
@@ -239,153 +238,80 @@ public static function removeDiagnostic(core_kernel_classes_Resource $testCenter
-            ];
-            $history[] = [
-                'timestamp' => $startTime + rand(2010, 2020),
-                'session' => $sessionUri,
-                'role' => 'Test Taker',
-                'actor' => 'Billy LaPorte',
-                'event' => 'start item',
-                'details' => '',
-                'context' => 'Section 1 - item 3/3'
-            ];
-            $history[] = [
-                'timestamp' => $startTime + rand(2040, 2050),
-                'session' => $sessionUri,
-                'role' => 'Proctor',
-                'actor' => 'Proctor 1',
-                'event' => 'terminate',
-                'details' => ['Fire emergency', 'Exit code: T'],
-                'context' => 'Section 1 - item 3/3'
-            ];
+            $valid = false;
+            $deliveryExecution = \taoDelivery_models_classes_execution_ServiceProxy::singleton()->getDeliveryExecution($sessionUri);
+            $delivery = $deliveryExecution->getDelivery();
+            $executions = $deliveryService->getCurrentDeliveryExecutions($delivery, $testCenter, $options);
+            foreach($executions as $execution){
+                if($execution['delivery_execution_id'] === $sessionUri && $execution['authorized_by'] === $proctor->getUri()){
+                    $valid = true;
+                    continue;
+                }
+            }
+            if(!$valid){
+                continue;
+            }
+
+            if($logHistory){
+                $deliveryLog->log($deliveryExecution->getIdentifier(), 'HISTORY', array());
+            }
+
+            $logs = $deliveryLog->get($deliveryExecution->getIdentifier());
+            $exportable = array();
+            foreach($logs as $data){
+                if($data['event_id'] !== 'HEARTBEAT' && strpos($data['event_id'], 'latency') === FALSE){
+                    $author = new \core_kernel_classes_Resource($data['created_by']);
+                    $proctorRole = new \core_kernel_classes_Resource('http://www.tao.lu/Ontologies/TAOProctor.rdf#ProctorRole');
+                    $role = ($userService->userHasRoles($author, $proctorRole)) ? __('Proctor') : __('Test-Taker');
+
+                    if(is_string($data['data'])){
+                        $data['data'] = json_decode($data['data'], true);
+                    }
+                    if(isset($data['data']['type'])) {
+                        $event_id = $data['data']['type'];
+
+                        $context = (isset($data['data']['context']['readable']))?$data['data']['context']['readable'] : '';
+                        $details = (isset($data['data']['context']['shortcut']))?$data['data']['context']['shortcut']: '';
+                    } else {
+                        $details = (isset($data['data']['reason']['reasons']) && !is_null($data['data']['reason']['reasons']))?array_merge(array_values($data['data']['reason']['reasons']), array($data['data']['reason']['comment'])) : '';
```

Comment:
Is the 'reason' index always exist if the type is not provided ? This could lead to issue if the data is extended later. I would rather place a ` elseif (isset($data['data']['reason']))` here. Or whatever could ensure the needed format.